test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,222 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidSerializer() = default;
+	~TestUuidSerializer() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+// Test string serialization
+TEST_F(TestUuidSerializer, SerializeToString) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABCDEF);
+	uuid.set_lsb(0xFEDCBA0987654321);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "12345678-90ab-cdef-fedc-ba0987654321");
+}
+
+// Test serialization with leading zeros in each segment
+TEST_F(TestUuidSerializer, SerializeWithLeadingZeros) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234007800AB);
+	uuid.set_lsb(0x00FE00BA09876543);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "00001234-0078-00ab-00fe-00ba09876543");
+}
+
+// Test serialization with mixed case letters
+TEST_F(TestUuidSerializer, SerializeWithMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABcDEF);
+	uuid.set_lsb(0x00dcbA0987654321);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value in lowercase
+	EXPECT_EQ(serialized_uuid, "12345678-90ab-cdef-00dc-ba0987654321");
+}
+
+// Test serialization with leading zeros and mixed case letters
+TEST_F(TestUuidSerializer, SerializeWithLeadingZerosAndMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234567890AB);
+	uuid.set_lsb(0xFedcba0987654982);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "00001234-5678-90ab-fedc-ba0987654982");
+}
+
+// Test serialization with leading/trailing zeros and mixed case letters
+TEST_F(TestUuidSerializer,
+       SerializeWithLeadingZerosAndTrailingZerosAndMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234567890AB);
+	uuid.set_lsb(0xFedcba0987600000);
```

Comment:
The mixed case letters should be in a string that is deserialized. Otherwise this test case is perfect for the leading zeros.